lib/client-assets.php

```diff
@@ -339,92 +342,106 @@ function gutenberg_register_scripts_and_styles() {
-
-		wp_enqueue_script(
-			'gutenberg-live-reload',
-			$live_reload_url
-		);
-	}
+	$styles->add_data( 'wp-block-directory', 'rtl', 'replace' );
 }
-add_action( 'wp_enqueue_scripts', 'gutenberg_register_scripts_and_styles', 5 );
-add_action( 'admin_enqueue_scripts', 'gutenberg_register_scripts_and_styles', 5 );
+add_action( 'wp_default_styles', 'gutenberg_register_packages_styles' );
 
 /**
- * Registers vendor JavaScript files to be used as dependencies of the editor
+ * Registers common scripts and styles to be used as dependencies of the editor
  * and plugins.
  *
- * This function is called from a script during the plugin build process, so it
- * should not call any WordPress PHP functions.
- *
  * @since 0.1.0
  */
-function gutenberg_register_vendor_scripts() {
-	$suffix = SCRIPT_DEBUG ? '' : '.min';
-
-	// Vendor Scripts.
-	$react_suffix = ( SCRIPT_DEBUG ? '.development' : '.production' ) . $suffix;
+function gutenberg_enqueue_block_editor_assets() {
+	global $wp_scripts;
 
-	// TODO: Overrides for react, react-dom and lodash are necessary
-	// until WordPress 5.3 is released.
-	gutenberg_register_vendor_script(
-		'react',
-		'https://unpkg.com/react@16.9.0/umd/react' . $react_suffix . '.js',
-		array( 'wp-polyfill' )
-	);
-	gutenberg_register_vendor_script(
-		'react-dom',
-		'https://unpkg.com/react-dom@16.9.0/umd/react-dom' . $react_suffix . '.js',
-		array( 'react' )
+	wp_add_inline_script(
+		'wp-api-fetch',
+		sprintf(
+			'wp.apiFetch.nonceMiddleware = wp.apiFetch.createNonceMiddleware( "%s" );' .
+			'wp.apiFetch.use( wp.apiFetch.nonceMiddleware );' .
+			'wp.apiFetch.nonceEndpoint = "%s";',
+			( wp_installing() && ! is_multisite() ) ? '' : wp_create_nonce( 'wp_rest' ),
+			admin_url( 'admin-ajax.php?action=gutenberg_rest_nonce' )
+		),
+		'after'
 	);
-	gutenberg_register_vendor_script(
-		'lodash',
-		'https://unpkg.com/lodash@4.17.15/lodash' . $suffix . '.js'
+
+	// TEMPORARY: Core does not (yet) provide persistence migration from the
+	// introduction of the block editor and still calls the data plugins.
+	// We unset the existing inline scripts first.
+	$wp_scripts->registered['wp-data']->extra['after'] = array();
+	wp_add_inline_script(
+		'wp-data',
+		implode(
+			"\n",
+			array(
+				'( function() {',
+				'	var userId = ' . get_current_user_ID() . ';',
+				'	var storageKey = "WP_DATA_USER_" + userId;',
+				'	wp.data',
+				'		.use( wp.data.plugins.persistence, { storageKey: storageKey } );',
+				'	wp.data.plugins.persistence.__unstableMigrate( { storageKey: storageKey } );',
+				'} )();',
+			)
+		)
 	);
+
+	if ( defined( 'GUTENBERG_LIVE_RELOAD' ) && GUTENBERG_LIVE_RELOAD ) {
+		$live_reload_url = ( GUTENBERG_LIVE_RELOAD === true ) ? 'http://localhost:35729/livereload.js' : GUTENBERG_LIVE_RELOAD;
+
+		wp_enqueue_script(
+			'gutenberg-live-reload',
+			$live_reload_url
+		);
+	}
 }
+add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_block_editor_assets', 5 );
```

Comment:
I don't think we want to have those assets registered with `wp_enqueue_scripts` and `admin_enqueue_scripts`. We probably can safely remove 5 priority as well. @youknowriad and @mcsf, can you double-check?

Reply:
I removed the priority and Travis doesn't seem to care 🤷‍♂

Reply:
Yeah, should work, ideally.